qintelqwatch_connector.py

```diff
@@ -0,0 +1,412 @@
+
+    def _validate_integer(self, parameter, key):
+        """
+        Validate an integer.
+
+        :param action_result: Action result or BaseConnector object
+        :param parameter: input parameter
+        :param key: input parameter message key
+        :return: status phantom.APP_ERROR/phantom.APP_SUCCESS, integer value of the parameter or None in case of failure
+        """
+        if parameter is not None:
+            try:
+                if not float(parameter).is_integer():
+                    return self.set_status(phantom.APP_ERROR, VALID_INT_MSG.format(param=key)), None
+
+                parameter = int(parameter)
+            except:
+                return self.set_status(phantom.APP_ERROR, VALID_INT_MSG.format(param=key)), None
+
+            if parameter < 0:
+                return self.set_status(phantom.APP_ERROR, NON_NEG_INT_MSG.format(param=key)), None
+
+        return phantom.APP_SUCCESS, parameter
+
+    def _handle_test_connectivity(self):
+
+        action_result = self.add_action_result(ActionResult())
+        try:
+            res = search_qwatch(None, None, 'ping', **self.client_args)
+            self.debug_print(f'qwatch test connectivity return: {res}')
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR,
+                            f'Test Connectivity Failed {str(e)}')
+
+        self.save_progress("Test Connectivity Passed")
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def _qwatch_query(self, search_term, search_type, query_type, search_args):
+
+        kwargs = deepcopy(self.client_args)
+        kwargs.update(search_args)
+
+        try:
+            return search_qwatch(
+                search_term,
+                search_type,
+                query_type,
+                **kwargs
+            )
+        except Exception as e:
+            self.debug_print(f'qwatch lookup failed: {str(e)}')
+            raise Exception(str(e))
+
+    def _make_qwatch_times(self, params):
+
+        # stored - start time
+        start_time = self._state.get('last_poll')
+        if start_time:
+            start_time = datetime.strptime(start_time, '%Y-%m-%dT%H:%M:%SZ')
+            start_time = int(start_time.timestamp())
+
+        # override - start time
+        QWATCH_START = os.getenv('QWATCH_START')
+        if QWATCH_START:
+            start_time = int(QWATCH_START)
+            self.debug_print('_make_qwatch_times',
+                             f'overriding start_time to {start_time}')
+
+        # initial poll - start time
+        if not start_time:
+            if not (1 <= self.qwatch_initial_window <= 90):
+                raise Exception(
+                    'Config Error: QWatch Initial Ingestion Window '
+                    'parameter must be between 1 and 90'
+                )
+
+            start_time = datetime.utcnow() - \
+                         timedelta(days=self.qwatch_initial_window)
```

Comment:
Can this be on the same line? If not, what do you think about:
```python
start_time = (
    datetime.utcnow() - timedelta(days=self.qwatch_initial_window))
```

Reply:
Done